### PR TITLE
[serve] Fully remove the v1 (`SINGLE_APP`) REST API

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

This API has previously been marked deprecated and removed from the documentation. This PR fully removes it.

Minor follow-up issue discovered in the process of updating the tests: https://github.com/ray-project/ray/issues/40907

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by GeneDer — APPROVED** [review]
:shipit:

File: python/ray/serve/tests/test_deploy_app.py [hunk 1]
```diff
@@ -755,48 +549,6 @@ def test_controller_recover_and_deploy(client: ServeControllerClient):
     assert SERVE_DEFAULT_APP_NAME not in serve.status().applications
 
 
-@pytest.mark.parametrize(
-    "field_to_update",
-    ["import_path", "runtime_env", "ray_actor_options"],
-)
-def test_deploy_config_update_heavyweight(
```
Comment (zcin) [inline python/ray/serve/tests/test_deploy_app.py:762]: Why was this test removed? I think this functionality should still be tested

File: python/ray/serve/tests/test_deploy_app.py [hunk 2]
```diff
@@ -1238,27 +956,6 @@ def serve_running():
     wait_for_condition(serve_running)
 
 
-def test_deployments_not_listed_in_config(client: ServeControllerClient):
```
Comment (zcin) [inline python/ray/serve/tests/test_deploy_app.py:1241]: Same for this test

**Review by zcin — APPROVED** [review]
LGTM, just flagged two tests that were removed

File: python/ray/serve/schema.py [hunk 3]
```diff
@@ -984,10 +933,10 @@ class ServeInstanceDetails(BaseModel, extra=Extra.forbid):
         )
     )
     deploy_mode: ServeDeployMode = Field(
+        default=ServeDeployMode.MULTI_APP,
         description=(
-            "Whether a single-app config of format ServeApplicationSchema or multi-app "
-            "config of format ServeDeploySchema was deployed to the cluster."
-        )
+            "[DEPRECATED]: single-app configs are removed, so this is always MULTI_APP."
```
[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/serve/schema.py:938]: yea let me add it

**Pull request merged**